src/Microsoft.SqlTools.ServiceLayer/EditData/UpdateManagement/RowEditBase.cs

```diff
@@ -45,8 +45,10 @@ protected RowEditBase(long rowId, ResultSet associatedResultSet, EditTableMetada
             }
 
             RowId = rowId;
-            AssociatedResultSet = associatedResultSet;
             AssociatedObjectMetadata = associatedMetadata;
+            AssociatedResultSet = associatedResultSet;
+
+            AssociatedObjectMetadata.Columns = EditTableMetadata.FilterColumnMetadata(AssociatedObjectMetadata.Columns, AssociatedResultSet.Columns);
```

Comment:
AFAICT, this is already being done when the associated metadata is extended. If we remove this operation, then we can make the FilterColumnMetadata a private helper method.